controllers/reconcilers/kserve_metrics_dashboard_reconciler.go

```diff
@@ -0,0 +1,202 @@
+	deltaProcessor   processors.DeltaProcessor
+}
+
+func NewKserveMetricsDashboardReconciler(client client.Client) *KserveMetricsDashboardReconciler {
+	return &KserveMetricsDashboardReconciler{
+		client:           client,
+		telemetryHandler: resources.NewConfigMapHandler(client),
+		deltaProcessor:   processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *KserveMetricsDashboardReconciler) Reconcile(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) error {
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource(ctx, log, isvc)
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(ctx, log, desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *KserveMetricsDashboardReconciler) createDesiredResource(ctx context.Context, log logr.Logger, isvc *kservev1beta1.InferenceService) (*corev1.ConfigMap, error) {
+
+	// resolve SR
+	isvcRuntime := isvc.Spec.Predictor.Model.Runtime
+	runtime := &kservev1alpha1.ServingRuntime{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: *isvcRuntime, Namespace: isvc.Namespace}, runtime); err != nil {
+		log.Error(err, "Could not determine servingruntime for isvc")
+	}
+
+	servingRuntimeImage := runtime.Spec.Containers[0].Image
+	re := regexp.MustCompile(`/([^/@]+)[@:]`)
+	findImageName := re.FindStringSubmatch(servingRuntimeImage)
+	servingRuntime := findImageName[1]
+
+	var data []byte
+	switch servingRuntime {
+	case "openvino_model_server":
+		if ovmsData == nil {
+			ovmsData, err := os.ReadFile("ovms-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+			}
+			data = ovmsData
+		}
+	case "text-generation-inference":
+		if tgisData == nil {
+			tgisData, err := os.ReadFile("tgis-metrics.json")
+			if err != nil {
+				log.Error(err, "Unable to load metrics dashboard template file:", err)
+			}
+			data = tgisData
+		}
+	}
+
+	finaldata := substituteVariablesInQueries(string(data), isvc.Namespace, isvc.Name)
+	// Create ConfigMap object
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      isvc.Name + "-metrics-dashboard",
+			Namespace: isvc.Namespace,
+		},
+		Data: map[string]string{
+			"metrics": finaldata,
+		},
+	}
+	// Add labels to the configMap
+	configMap.Labels = map[string]string{
+		"app.opendatahub.io/kserve": "true",
+	}
+	if err := ctrl.SetControllerReference(isvc, configMap, r.client.Scheme()); err != nil {
+		log.Error(err, "Unable to add OwnerReference to the Metrics Dashboard Configmap")
+		return nil, err
+	}
+
+	return configMap, nil
+}
+
+func substituteVariablesInQueries(data string, namespace string, name string) string {
+	return strings.ReplaceAll(strings.ReplaceAll(data, "${namespace}", namespace), "${model_name}", name)
```

Comment:
in the files proposed here https://github.com/opendatahub-io/odh-model-controller/pull/209/files these two envs are uppercase.